tests/benchmark/conftest.py

```diff
@@ -7,6 +7,17 @@
 
 def pytest_collection_modifyitems(config, items):
     """Add the `benchmark` marker to all tests under `./tests/benchmark`."""
+    marker_expr = config.getoption("-m", default="")
+    run_benchmarks = marker_expr and (
+        "benchmark" in marker_expr and "not benchmark" not in marker_expr
+    )
+
+    items_to_remove = []
     for item in items:
         if Path(__file__).parent in Path(item.fspath).parents:
             item.add_marker(pytest.mark.benchmark)
+            if not run_benchmarks:
```

Comment:
We could even also do the opposite on top of this: I.e. if we are running in benchmark, remove the tests that are not marked as benchmark tests.